src/utils/form/checkRecordValidation.ts

```diff
@@ -32,13 +32,31 @@ const addChoicesToGraphQLQuestions = (structure: any, newData: any) => {
     pages: structure.pages.map((page) => ({
       ...page,
       elements: page.elements.map((element) => {
-        if (element.gqlUrl) {
-          return {
-            ...element,
-            choices: newData[element.name].map((choice) => {
-              return { text: choice, value: choice };
-            }),
-          };
+        if (element.gqlUrl && newData[element.name]) {
+          if (['dropdown', 'radiogroup'].includes(element.type)) {
```

Comment:
@MwanPygmay @Matthis-M-ReliefApps 

FYI
this is what created the PROD issue of some records not being editable

we didn't tested enough this code change
indeed, the custom code for graphql would not support single select type, and would not cover cases where the value does not exist in the data yet